### ISSUE TITLE
Ajax request without content in response and not create ajax viex

### ISSUE THREAD

**Issue by JeanValJeann** [dereuromark/cakephp-ajax#23]
hi there,

I started to use this nice one Plugin but I didn't find out how to do an Ajax Request but not have to create ajax/view.ctp. I wondering if it's possible to make an Ajax Request wihtout content field in json returned and not have to create ajax/view.ctp at each ajax request through the Plugin.

Even when calling in Ajax an url having json extension, the plugin wants me to create a view.

**Comment by dereuromark** [comment]
Did u try serialize key?

**Comment by dereuromark** [comment]
See https://github.com/dereuromark/cakephp-ajax/blob/master/src/Controller/Component/AjaxComponent.php#L153

**Comment by JeanValJeann** [comment]
Yes, I serialized one key but I still have an error message wanting me to create an ajax view.
I tested to put the viewClass at false too.

Flash message et redirect is correctly handled

But now, I am trying to do an Ajax Request without having to create a view and by the way do not get back content field corresponding to the view. But I can't.

**Comment by dereuromark** [comment]
I see. In this case maybe we have to find a way to deactivate the view rendering.
Ideas welcome.

**Comment by JeanValJeann** [comment]
No rendering view if _serialize used within Controller seems to fit the enhancement.
Im gonna take a lookt a this, Im newbie in this kind of great Plugin. Im gonna take a look at a purpose you a pull request.

**Comment by dereuromark** [comment]
PR open: https://github.com/dereuromark/cakephp-ajax/pull/25

**Comment by JeanValJeann** [comment]
https://github.com/dereuromark/cakephp-ajax/blob/master/src/View/AjaxView.php#L105

What if change $this->_getViewFileName($view) condition (returning MissingTemplateException in case of View file Name doesn't exist) to a private function checking if the View file name exists returning boolean, then if the view file name do not exist we don't get MissingTemplateException and don't call $dataToSerialize['content'] = parent::render($view, $layout);, content key staying null. And if the View file name exsits we proceed to $dataToSerialize['content'] = parent::render($view, $layout); ?

In this way of doing, we avoid an MissingTemplateException throwing in case ajax/view.ctp not create and we can go on to use as usually the plugin. It'll just help to don"t create view file for action that will don't use it at all.

**Comment by dereuromark** [comment]
I added an actions whitelist and also tried to document a bit more explicitly the use cases.
Why do you want to use Ajax plugin instead of just JsonView for your use case? You dont seem to want to render any HTML snippets or re-use an existing non AJAX action, right?
